src/effects/VST3/internal/x11/SocketWindow.cpp

```diff
@@ -0,0 +1,44 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file SocketWindow.cpp
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+#include "SocketWindow.h"
+#include "RunLoop.h"
+#include "PlugFrame.h"
+
+using namespace internal::x11;
+
+void SocketWindow::OnMap(GtkWidget* widget, gpointer data)
+{
+   using namespace Steinberg;
+
+   auto self = reinterpret_cast<SocketWindow*>(data);
+
+   static auto runLoop = [&]() {
+      auto display = GDK_WINDOW_XDISPLAY(gtk_widget_get_window(widget));
+      return owned(safenew RunLoop(display));
+   }();
+
+   auto frame = owned(safenew PlugFrame(runLoop.get(), self->GetParent()));
+   if(self->mPlugView->setFrame(frame) == Steinberg::kResultOk)
+   {
+      self->mPlugView->attached(
+         (void*)gtk_socket_get_id(GTK_SOCKET(widget)), 
+         Steinberg::kPlatformTypeX11EmbedWindowID
+      );
+   }
```

Comment:
The smart pointer runLoop is destroyed here.  But runLoop.get() was passed to the PlugFrame constructor.  Is there a dangling pointer danger?

Reply:
I'm not sure that understand your question.
The type of `runLoop` variable is `IPtr<RunLoop>`. `owned` is a macro which grabs raw pointer and constructs IPtr object without reference count increment. `runLoop.get()` returns a raw pointer, but `RunLoop` constructor assigns it to the field of type IPtr<IRunLoop>

Reply:
I'm not sure that understand your question.
The type of `runLoop` variable is `IPtr<RunLoop>`. `owned` is a macro which grabs raw pointer and constructs IPtr object without reference count increment. `runLoop.get()` returns a raw pointer, but `RunLoop` constructor assigns it to the field of type IPtr<IRunLoop>